tsl/src/remote/dist_ddl.c

```diff
@@ -1070,6 +1066,70 @@ dist_ddl_get_analyze_stats(const ProcessUtilityArgs *args)
 	chunk_api_update_distributed_hypertable_stats(relid);
 }
 
+/*
+ * Check if it is allowed to execute distributed DDL operation on a
+ * global object such as SCHEMA, DATABASE, etc.
+ */
+static bool
+dist_ddl_process_global_object(ProcessUtilityArgs *args)
+{
+	if (ts_guc_enable_ddl_on_global_objects)
+		return true;
+
+	/* disable this functionality while any extension being
+	 * created or upgraded */
+	if (creating_extension)
+		return false;
+
+	switch (nodeTag(args->parsetree))
+	{
+		case T_CreateSchemaStmt:
+		case T_DropOwnedStmt:
+		case T_ReassignOwnedStmt:
+		case T_AlterDefaultPrivilegesStmt:
+			return false;
+
+		case T_DropStmt:
+		{
+			const DropStmt *stmt = castNode(DropStmt, args->parsetree);
+			if (stmt->removeType == OBJECT_SCHEMA)
+				return false;
```

Comment:
So this kind of duplicates the logic in `dist_ddl_drop`? I wonder if we should somehow keep it to one place to avoid confusion.

Reply:
I don't think this is related to the `dist_ddl_drop` because it is needed to be scheduled first which is done by the preprocess function first.

But yes, it is a bit of logic duplication in a way that it could be checked separately for each function. But in such case it slightly complicates overall logic, because each function would require separately to check this guc and would require another way to see if the query been skipped for testing.

Anyway, if people are feel strongly about it, I can blend this check to the processing functions.